### ISSUE TITLE
Implementation of Additive Attention

### ISSUE THREAD

**Issue by BingzhaoZhu** [jrzaurin/pytorch-widedeep#110]
Hi,

I have a question about the implementation of FastFormer. [The current implementation in WideDeep calculated the attention weight and "softmax" across the last dimension, which correspond to "num_heads". ](https://github.com/jrzaurin/pytorch-widedeep/blob/master/pytorch_widedeep/models/tabular/transformers/_attention_layers.py#:~:text=alphas%20%3D%20(self.W_q(q)%20/%20math.sqrt(self.head_dim)).softmax(dim%3D%2D1)).

In the [paper](https://arxiv.org/pdf/2108.09084.pdf), the "softmax" were performed over the "num_tokens" (e.g., sequence length) dimension, see equation (3) (5). This issue will cause very large vaules in "global_query" when the sequence is very long and NaNs in the output . Should it just be ".softmax(dim=1)"?

Thanks for your attention

**Comment by jrzaurin** [comment]
Hey @BingzhaoZhu 

I think you are right, this is a bug, thanks for having a look to the code in such detail! 

Let me fix it and I will let you know when is done

**Comment by jrzaurin** [comment]
@BingzhaoZhu 

It is not fixed in master and publish to pypi

Thanks again for raising the issue :)
